chore: add missing return types and return statement

juantejer4 opened the pull request:
# ⚡ Add missing return types to Laravel boilerplate stubs - [86dr3nyh6](https://app.clickup.com/t/86dr3nyh6) ⚡

## 💻 What type of change is this?

- [x] 💎 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Styling
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build / CI

## ⭐ Description

<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
Example:
-->

I've added return types  for all stub methods, and each including a return statement

## 💬 Comments

## ✅ Checklist

- [x] This PR can be merged (it is not a draft, work in progress, or blocked on another PR)
- [ ] I have tested this change locally in multiple screen sizes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

juantejer4 committed:
feat: add missing return types and return statement

Commit:
feat: add replacements to `ControllerMakeCommand`

gianfranco-rocco commented on stubs/controller.stub:
```diff
@@ -2,44 +2,61 @@
+    public function create(): JsonResponse
     {
-        //
+        return responder()
+            ->success()
+            ->respond();
```
```suggestion
        return View::make('create');
```

gianfranco-rocco commented on stubs/controller.stub:
```diff
@@ -2,44 +2,61 @@
+    public function show(string $id): View
     {
-        //
+        return responder()
+            ->success()
+            ->respond();
```
```suggestion
        return View::make('show');
```

gianfranco-rocco commented on stubs/controller.stub:
```diff
@@ -2,44 +2,61 @@
     }
 
-    public function edit(string $id)
+    public function edit(string $id): JsonResponse
```
```suggestion
    public function edit(string $id): ViewContract
```

gianfranco-rocco commented on stubs/controller.stub:
```diff
@@ -2,44 +2,61 @@
+    public function edit(string $id): JsonResponse
     {
-        //
+        return responder()
+            ->success()
+            ->respond();
```
```suggestion
        return View::make('edit');
```

gianfranco-rocco commented on stubs/controller.stub:
```diff
@@ -2,44 +2,61 @@
-        //
+        return responder()
+            ->success()
+            ->respond();
```
```suggestion
            ->respond(JsonResponse::HTTP_CREATED);
```

gianfranco-rocco commented on stubs/controller.stub:
```diff
@@ -2,44 +2,61 @@
-        //
+        return responder()
+            ->success()
+            ->respond();
```
```suggestion
            ->respond(JsonResponse::HTTP_NO_CONTENT);
```

gianfranco-rocco commented on stubs/factory.stub:
This stub's previous return statement was right. This is the stub for factories, not controllers. Changes should be reverted.

gianfranco-rocco commented on stubs/job.stub:
Revert changes.

gianfranco-rocco commented on stubs/mail.stub:
Revert changes.

gianfranco-rocco commented on stubs/markdown-mail.stub:
Revert changes.

gianfranco-rocco commented on stubs/markdown-notification.stub:
Revert changes.

gianfranco-rocco commented on stubs/notification.stub:
Revert changes.

gianfranco-rocco commented on stubs/notification.stub:
Revert changes.

gianfranco-rocco commented on stubs/policy.stub:
Revert changes.

gianfranco-rocco commented on stubs/request.stub:
Revert changes.

gianfranco-rocco commented on stubs/resource-collection.stub:
Revert changes.

gianfranco-rocco commented on stubs/resource.stub:
Revert changes.

gianfranco-rocco commented on stubs/view-component.stub:
Revert changes.

juantejer4 committed:
feat: add several suggestions from code review

Co-authored-by: Gianfranco Rocco <110541583+gianfranco-rocco@users.noreply.github.com>

juantejer4 committed:
fix: revert unnecessary changes

Commit:
fix: `View` import

Commit:
feat: override `buildFormRequestReplacements`

Commit:
refactor: spread operator

Commit:
feat: replace base `Request` with `FormRequest`

Commit:
feat: pass parent model to view

Commit:
fix: namespace/import order

Commit:
fix: `mai.stub` return statements

Commit:
style: indentation

Commit:
style: ascending import order

gianfranco-rocco commented:
@sojeda, I need your experience to correct this PR.

Commit:
chore: various changes

Replaced `abort()` with `App::abort()`.

Removed message from `store` & `update` responses.

Deleted `make:controller` command override.

sojeda commented on stubs/controller.nested.singleton.api.stub:
```diff
@@ -4,29 +4,35 @@ declare(strict_types=1);
     public function destroy({{ parentModel }} ${{ parentModelVariable }}): never
     {
-        abort(404);
+        App::abort(JsonResponse::HTTP_NOT_FOUND);
```
```
return responder()
            ->error()
            ->respond(JsonResponse::HTTP_NOT_FOUND);
```

Commit:
feat: respond with not found instead of aborting
